tests/foreman/api/test_audit.py

```diff
@@ -0,0 +1,175 @@
+            entity_type = entity_item.get(
+                'entity_type', created_entity.__class__.__name__.lower())
+            if entity_type == 'user':
+                entity_value = created_entity.login
+            else:
+                entity_value = created_entity.name
+            value_template = entity_item.get('value_template', None)
+            if value_template is not None:
+                entity_value = value_template.format(created_entity.name)
+            if entity_type == 'os':
+                entity_value += ' {}'. format(created_entity.major)
+            if entity_type == 'subnet':
+                entity_value += ' ({}/{})'. format(
+                    created_entity.network, created_entity.cidr
+                )
+            audit = entities.Audit().search(
+                query={'search': 'type={0}'.format(entity_type)})[0]
+            self.assertEqual(audit.auditable_name, entity_value)
+            self.assertEqual(audit.auditable_id, created_entity.id)
+            self.assertEqual(audit.action, 'create')
+            self.assertEqual(audit.version, 1)
+
+    @tier1
+    def test_positive_update_by_type(self):
+        """Update some entities of different types and check audit logs for
+        these events using entity type as search criteria
+
+        :id: 43e73a11-b241-4b91-bdf6-e966366014e8
+
+        :expectedresults: Audit logs contain corresponding entries per each
+            update event
+        """
+        for entity in [
+            entities.Architecture(),
+            entities.Domain(),
+            entities.HostGroup(),
+            entities.Location(),
+            entities.Organization(),
+            entities.Role(),
+            entities.UserGroup(),
+        ]:
+            created_entity = entity.create()
+            name = created_entity.name
+            new_name = gen_string('alpha')
+            created_entity.name = new_name
+            created_entity = created_entity.update(['name'])
+            audit = entities.Audit().search(
+                query={'search': 'type={0}'.format(
+                    created_entity.__class__.__name__.lower())
+                }
+            )[0]
+            self.assertEqual(audit.auditable_name, name)
+            self.assertEqual(audit.auditable_id, created_entity.id)
+            self.assertEqual(
+                audit.audited_changes['name'], [name, new_name])
+            self.assertEqual(audit.action, 'update')
+            self.assertEqual(audit.version, 2)
+
+    @tier1
+    def test_positive_delete_by_type(self):
+        """Delete some entities of different types and check audit logs for
+        these events using entity type as search criteria
+
+        :id: de9b056f-10da-485a-87ce-b02a9efff15c
+
+        :expectedresults: Audit logs contain corresponding entries per each
+            delete event
+        """
+        for entity in [
+            entities.Architecture(),
```

Comment:
That's probably a nitpick, but worth mentioning - better to initialize specific entity inside the loop body instead of initializing all the entities at the very beginning of the test. 
I mean
```python
for entity in [
    entities.Architecture,
    ...
]:
    created_entity = entity().create()
```
You know the reasons - better memory management, faster execution time (if the failure happens on one of the first entities, others won't be even initialized). For some 'heavier' things like VMs that'd be a blocker comment, but of course, in this particular case, entities initialization is very fast operation, so that's just a nitpick :)
Applicable for all the tests.